lib/messages/search_request.js

```diff
@@ -105,7 +105,13 @@ SearchRequest.prototype._parse = function (ber) {
 SearchRequest.prototype._toBer = function (ber) {
   assert.ok(ber)
 
-  ber.writeString(this.baseObject.toString())
+  // Format only with commas, since that is what RFC 4514 mandates.
+  // There's a gotcha here: even though it's called baseObject,
+  //  it can be a string or a DN object.
+  var formattedDN = dn.DN.isDN(this.baseObject)
```

Comment:
I think we should fork `node-asn1` into the org after publishing `ldapjs@2` and potentially clean up ambiguous things like this.

Reply:
@jsumners Does `node-asn1` actually have anything to do with this? It's not a bad idea in general, just missing the idea as it regards DN formatting.

Reply:
It's the module that provides BER encoding.